File: src/leetcode/check-if-word-equals-summation-of-two-words/check-if-word-equals-summation-of-two-words.js
/**
 * Leetcode #1880
 * Difficulty: Easy
 * https://leetcode.com/problems/check-if-word-equals-summation-of-two-words/
 */
/**
 * @param {string} firstWord
 * @param {string} secondWord
 * @param {string} targetWord
 * @return {boolean}
 */
export const isSumEqual = (firstWord, secondWord, targetWord) => {
  let first = '';
  let second = '';
  let target = '';

  for (let i = 0; i < firstWord.length; i += 1) {
    const code = firstWord[i].charCodeAt() - 97;
    first += code;
  }

  for (let i = 0; i < secondWord.length; i += 1) {
    const code = secondWord[i].charCodeAt() - 97;
    second += code;
  }

  for (let i = 0; i < targetWord.length; i += 1) {
    const code = targetWord[i].charCodeAt() - 97;
    target += code;
  }

  return Number(first) + Number(second) === Number(target);
};

File: src/leetcode/check-if-word-equals-summation-of-two-words/check-if-word-equals-summation-of-two-words.rs
impl Solution {
    pub fn is_sum_equal(first_word: String, second_word: String, target_word: String) -> bool {
        Self::str_sum(first_word) + Self::str_sum(second_word) == Self::str_sum(target_word)
    }

    fn str_sum(word: String) -> u8 {
        let mut num: u8 = 0;

        for c in word.chars() {
            let c_num = c as u8 - 'a' as u8;
            num = num * 10 + c_num;
        }

        num
    }
}